src/components/dashboard/FeedModalList.js

```diff
@@ -72,49 +75,67 @@ const FeedModalList = ({
       // https://stackoverflow.com/questions/40200660/react-native-scrollto-with-interactionmanager-not-working
       setTimeout(() => {
         flatListRef && flatListRef.current && flatListRef.current.scrollToOffset({ animated: false, offset })
+        setLoading(false)
       }, 0)
     }
-  }, [offset, flatListRef])
-
-  const getItemLayout = (_: any, index: number) => {
-    const length = screenWidth
-    return { index, length, offset: length * index }
-  }
-
-  const renderItemComponent = ({ item, separators, index }: ItemComponentProps) => (
-    <View style={styles.horizontalListItem}>
-      <FeedModalItem
-        navigation={navigation}
-        item={item}
-        separators={separators}
-        fixedHeight
-        onPress={() => handleFeedSelection(item, false)}
-      />
-    </View>
+  }, [offset, flatListRef, setLoading])
+
+  const getItemLayout = useCallback(
+    (_: any, index: number) => {
+      const length = screenWidth
+      return { index, length, offset: length * index }
+    },
+    [screenWidth],
+  )
+
+  const renderItemComponent = useCallback(
+    ({ item, separators }: ItemComponentProps) => (
+      <View style={styles.horizontalListItem}>
+        <FeedModalItem
+          navigation={navigation}
+          item={item}
+          separators={separators}
+          fixedHeight
+          onPress={() => handleFeedSelection(item, false)}
+        />
+      </View>
+    ),
+    [handleFeedSelection, navigation],
+  )
+
+  const keyExtractor = useCallback(item => item.id || item.createdDate, [])
+
+  const initialNumToRender = useMemo(() => Math.abs(data.findIndex(item => item.id === selectedFeed.id)), [
+    selectedFeed,
+    data,
+  ])
+
+  const handleScroll = useCallback(
+    ({ nativeEvent }) => {
+      slideEvent()
```

Comment:
```suggestion
      slideEventRef.current()
```

Reply:
otherwise if you'll go to other screen then go back to dashboard and will start scrolling, the event won't trigger again